modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLErrorResponseTestCase.java

```diff
@@ -0,0 +1,307 @@
+        Map<String, String> paramters = new HashMap<String, String>();
+        paramters.put("SAMLRequest", samlRequest);
+        paramters.put("RelayState", RELAY_STATE);
+        response = sendSAMLMessage(SAML_SSO_URL, paramters);
+        EntityUtils.consume(response.getEntity());
+        response = sendRedirectRequest(response);
+
+        String sessionKey = extractDataFromResponse(response, "name=\"sessionDataKey\"", 1);
+        response = sendPOSTMessage(sessionKey);
+        EntityUtils.consume(response.getEntity());
+
+        String location = response.getFirstHeader("Location").getValue();
+        assertTrue(location.contains("RelayState=" + RELAY_STATE), "Redirection header to notification.do" +
+                " page should contain RelayState query param sent with the request");
+        assertTrue(location.contains("ACSUrl=" + URLEncoder.encode(String.format(ACS_URL,
+                ARTIFACT_ID), "UTF-8")), "Redirection header to notification.do" +
+                " page should contain ACS query param");
+    }
+
+
+    private void createApplication() throws Exception {
+
+        ServiceProvider serviceProvider = new ServiceProvider();
+        serviceProvider.setApplicationName(APPLICATION_NAME);
+        serviceProvider.setDescription("This is a test Service Provider");
+        applicationManagementServiceClient.createApplication(serviceProvider);
+        serviceProvider = applicationManagementServiceClient.getApplication(APPLICATION_NAME);
+
+        InboundAuthenticationRequestConfig requestConfig = new InboundAuthenticationRequestConfig();
+        requestConfig.setInboundAuthType(INBOUND_AUTH_TYPE);
+        requestConfig.setInboundAuthKey(ARTIFACT_ID);
+
+        InboundAuthenticationConfig inboundAuthenticationConfig = new InboundAuthenticationConfig();
+        inboundAuthenticationConfig.setInboundAuthenticationRequestConfigs(
+                new InboundAuthenticationRequestConfig[]{requestConfig});
+
+        serviceProvider.setInboundAuthenticationConfig(inboundAuthenticationConfig);
+        applicationManagementServiceClient.updateApplicationData(serviceProvider);
+    }
+
+    private Tomcat getTomcat() {
+
+        Tomcat tomcat = new Tomcat();
+        tomcat.getService().setContainer(tomcat.getEngine());
+        tomcat.setPort(DEFAULT_TOMCAT_PORT);
+        tomcat.setBaseDir("");
+
+        StandardHost stdHost = (StandardHost) tomcat.getHost();
+
+        stdHost.setAppBase("");
+        stdHost.setAutoDeploy(true);
+        stdHost.setDeployOnStartup(true);
+        stdHost.setUnpackWARs(true);
+        tomcat.setHost(stdHost);
+
+        setSystemProperties();
+        return tomcat;
+    }
+
+    private void setSystemProperties() {
+
+        URL resourceUrl = getClass().getResource(File.separator + "keystores" + File.separator
+                + "products" + File.separator + "wso2carbon.jks");
+        System.setProperty("javax.net.ssl.trustStore", resourceUrl.getPath());
```

Comment:
Isn't it nice to have constants for these values as you did in other methods, like:
private static final String JAVAX_NET_SSL_TRUSTORE = "javax.net.ssl.trustStore";

Reply:
Fixed with #2238